vehicle_e2e_test.go

```diff
@@ -0,0 +1,343 @@
+	expectedResponse := &Odometer{
+		Distance:        mockValue,
+		ResponseHeaders: s.responseHeaders,
+	}
+	mockURL := buildVehicleURL(string(OdometerPath), s.vehicle.id)
+	mockResponse := map[string]float64{"distance": mockValue}
+	mockVehicleAPI(mockURL, s.vehicle.accessToken, s.responseHeaders, mockResponse)
+
+	res, err := s.vehicle.GetOdometer(context.TODO())
+
+	assert.Nil(s.T(), err)
+	assert.Equal(s.T(), expectedResponse, res)
+}
+
+func (s *VehicleE2ETestSuite) TestGetOilE2E() {
+	mockLifeRemaining := 0.35
+	expectedResponse := &Oil{
+		LifeRemaining:   mockLifeRemaining,
+		ResponseHeaders: s.responseHeaders,
+	}
+	mockURL := buildVehicleURL(string(OilPath), s.vehicle.id)
+	mockResponse := map[string]interface{}{"lifeRemaining": mockLifeRemaining}
+	mockVehicleAPI(mockURL, s.vehicle.accessToken, s.responseHeaders, mockResponse)
+
+	res, err := s.vehicle.GetOil(context.TODO())
+
+	assert.Nil(s.T(), err)
+	assert.Equal(s.T(), expectedResponse, res)
+}
+
+func (s *VehicleE2ETestSuite) TestGetPermissionsE2E() {
+	mockPermissions := []string{"read_odometer", "read_batch"}
+	expectedResponse := &Permissions{
+		Permissions:     mockPermissions,
+		ResponseHeaders: s.responseHeaders,
+	}
+	mockURL := buildVehicleURL(string(PermissionsPath), s.vehicle.id)
+	mockResponse := map[string]interface{}{"permissions": mockPermissions}
+	mockVehicleAPI(mockURL, s.vehicle.accessToken, s.responseHeaders, mockResponse)
+
+	res, err := s.vehicle.GetPermissions(context.TODO())
+
+	assert.Nil(s.T(), err)
+	assert.Equal(s.T(), expectedResponse, res)
+}
+
+func (s *VehicleE2ETestSuite) TestGetTiresPressureE2E() {
+	mockFrontLeft := 219.0
+	mockFrontRight := 219.0
+	mockBackLeft := 219.0
+	mockBackRight := 219.0
+	expectedResponse := &TirePressure{
+		FrontLeft:       mockFrontLeft,
+		FrontRight:      mockFrontRight,
+		BackLeft:        mockBackLeft,
+		BackRight:       mockBackRight,
+		ResponseHeaders: s.responseHeaders,
+	}
+	mockURL := buildVehicleURL(string(TirePressurePath), s.vehicle.id)
+	mockResponse := map[string]interface{}{
+		"backLeft":   mockBackLeft,
+		"backRight":  mockBackRight,
+		"frontLeft":  mockFrontLeft,
+		"frontRight": mockFrontRight,
+	}
+	mockVehicleAPI(mockURL, s.vehicle.accessToken, s.responseHeaders, mockResponse)
+
+	res, err := s.vehicle.GetTiresPressure(context.TODO())
+
+	assert.Nil(s.T(), err)
+	assert.Equal(s.T(), expectedResponse, res)
+}
+
+func (s *VehicleE2ETestSuite) TestGetVINE2E() {
+	mockVIN := "1234DLFAJ4"
+	expectedResponse := &VIN{
+		VIN:             mockVIN,
+		ResponseHeaders: s.responseHeaders,
+	}
+	mockURL := buildVehicleURL(string(VINPath), s.vehicle.id)
+	mockResponse := map[string]interface{}{"vin": mockVIN}
+	mockVehicleAPI(mockURL, s.vehicle.accessToken, s.responseHeaders, mockResponse)
+
+	res, err := s.vehicle.GetVIN(context.TODO())
+
+	assert.Nil(s.T(), err)
+	assert.Equal(s.T(), expectedResponse, res)
+}
+
+func (s *VehicleE2ETestSuite) TestLockE2E() {
+	mockStatus := "1234DLFAJ4"
```

Comment:
Modify to be more realistic on what the status is.
```suggestion
	mockStatus := "success"
```